src/QueryFilterHelpers.ts

```diff
@@ -0,0 +1,604 @@
+ *
+ * @param inExpr The `in` expression to parse.
+ * @param library The library the expression resides in.
+ * @param parameters The parameters used for calculation.
+ * @returns Filter representation of the In clause.
+ */
+function interpretIn(inExpr: ELMIn, library: ELM, parameters: any): InFilter | DuringFilter | UnknownFilter {
+  let propRef: ELMProperty | null = null;
+  if (inExpr.operand[0].type == 'FunctionRef') {
+    propRef = interpretFunctionRef(inExpr.operand[0] as ELMFunctionRef);
+  } else if (inExpr.operand[0].type == 'Property') {
+    propRef = inExpr.operand[0] as ELMProperty;
+  } else if (inExpr.operand[0].type == 'End' || inExpr.operand[0].type == 'Start') {
+    const startOrEnd = inExpr.operand[0] as ELMUnaryExpression;
+    const suffix = startOrEnd.type === 'End' ? '.end' : '.start';
+    if (startOrEnd.operand.type == 'FunctionRef') {
+      propRef = interpretFunctionRef(startOrEnd.operand as ELMFunctionRef);
+    } else if (startOrEnd.operand.type == 'Property') {
+      propRef = startOrEnd.operand as ELMProperty;
+    }
+    if (propRef) {
+      propRef = {
+        type: 'Property',
+        path: propRef?.path + suffix,
+        scope: propRef?.scope,
+        localId: propRef?.localId,
+        locator: propRef?.locator,
+        source: propRef?.source
+      };
+    }
+  }
+
+  if (propRef == null) {
+    console.warn(
+      `could not resolve property ref for In:${inExpr.localId}. first operand is a ${inExpr.operand[0].type}`
+    );
+
+    const foundPropUsage = findPropertyUsage(inExpr, inExpr.localId);
+    if (foundPropUsage) {
+      console.warn(`  found property ref "${foundPropUsage.alias}.${foundPropUsage.attribute}" in first operand.`);
+      return foundPropUsage;
+    }
+
+    return { type: 'unknown' };
+  }
+
+  if (inExpr.operand[1].type == 'List') {
+    if (propRef.scope) {
+      return {
+        type: 'in',
+        alias: propRef.scope,
+        attribute: propRef.path,
+        valueList: (inExpr.operand[1] as ELMList).element.map(item => item.value) as (string | number)[],
+        localId: inExpr.localId
+      };
+    } else {
+      console.warn('Could not find scope for property reference');
+      return { type: 'unknown' };
+    }
+  } else if (inExpr.operand[1].type == 'Interval') {
+    const propRefInInterval = findPropertyUsage(inExpr.operand[1], undefined);
+    if (propRefInInterval) {
+      console.warn('cannot handle intervals constructed on query data right now');
+      return {
+        type: 'unknown',
+        localId: inExpr.localId,
+        alias: propRef.scope,
+        attribute: propRef.path
+      };
+    }
+
+    // build an expression that has the interval creation and
+    const intervalExpr = new cql.Expression({ operand: inExpr.operand[1] });
+    const ctx = new cql.PatientContext(new cql.Library(library), null, null, parameters);
+    const interval: cql.Interval = intervalExpr.arg.execute(ctx);
```

Comment:
Might be useful to stub this ad-hoc interval calculation into a helper function in case it needs to be repeated across other filters. Not sure if that would ever actually come up, but who knows

Reply:
Moved to its own function in ec0504d.